spec/plugin_mixin/http_client_spec.rb

```diff
@@ -54,4 +54,41 @@ class Dummy < LogStash::Inputs::Base
       include_examples("setting ca bundles", :truststore)
     end
   end
+
+  describe "with a client cert" do
+    let(:file) { Stud::Temporary.file }
```

Comment:
Any interest in validating that we provide good cert/key combination?

Flores::PKI can be used to create a cert for testing this if we need to. Maybe we can defer that for later - i'll file a ticket about it once this is merged.